### DISCUSSION TITLE
Retry closed PRs after "x" time automatically (opt-in)

### DISCUSSION THREAD

**Discussion by HonkingGoose** [renovatebot/renovate#13489] [Suggest an Idea]
## Intro

I was/am trying to figure out why Renovate bot re-opened some PRs, and was discussing the behavior with @Belco90, I said:

> I just checked the edit history of the Dependency Dashboard and cannot find a click from one of us. It's very weird that Renovate suddenly re-opens closed update PRs, normally they remain closed until we force them open via the dashboard...

@Belco90 then said:

> Perhaps it doesn't stop updating the dependency, but snoozes it so it reminds you to re-check it in the future?

It sounds a bit silly, but the more I thought about it, the more I thought there might be something in this idea... 😄 

## Current behavior

If you don't want a update PR from Renovate, you'll close it.
Then if you later do want the update, you'll need to:

- Change the PR title _or_
- Request a fresh PR from the Dependency Dashboard

## Rationale

Sometimes you're waiting on an external requirement, say you're proposed an update to ESLint `v8`, but your plugins only support ESlint `v7` for now, so you close the PR.

Now you need to keep track of the ESlint `v8` dependency yourself. But you might suspect that after 3 months the ESLint plugins will have updated to support ESLint `v8`. So you want to have a way to tell Renovate to try again after 3 months.

## Feasibility of concept

I don't think Renovate knows anything about _when_ it last ran? I think the idea is that it only knows about things that it can read from the GitHub repository, like branch name(s) and PR titles? I don't know if the idea of a "snooze" button/timer is even a thing that could be done with the current setup.

## Proposed behavior

If it would be possible, I imagine we'd keep the default behavior as-is, so "ignore PRs forevermore if closed". But people who want the bot to try again could set a "default timeout/snooze timer" by using presets, or their own custom rule(s).

We already use Bree.js to parse some human time frames, so we could maybe have presets like:

- `:prSnoozeOneMonth`: ignore closed PRs for a month, then re-open
- `:prSnoozeThreeMonths`: ignore closed PRs for 3 months, then re-open
- `:prSnoozeHalfYear`: ignore closed PRs for 6 months, then re-open

Even more mindblowing would be if you could tell Renovate when closing the PR: "Try again after x time", by clicking on a checkbox in the PR or Renovate comment, with some default time lengths.

## Discussion

What do you think about this whole concept?

**Comment by rarkins** [comment]
I don't like this idea :)

Given the dashboard, I think that's the best place/way to be reminded of closed PRs and recreate them

**Reply by HonkingGoose** [reply]
Fair enough. 😄 

Let's leave things as they are then.
